### ISSUE TITLE
Assistant : OpenAiApi.listRunSteps : @Body not supported on @Get

### ISSUE THREAD

**Issue by Aelentel** [TheoKanning/openai-java#411]
Hi, first and foremost, thanks everyone for this project :)

i encountered that issue when testing the Assistant API on v0.18.1

here's the exception (fluff removed) 

```
java.lang.IllegalArgumentException: Non-body HTTP method cannot contain @Body.
    for method OpenAiApi.listRunSteps
	at retrofit2.Utils.methodError(Utils.java:54)
	at retrofit2.Utils.methodError(Utils.java:43)
	at retrofit2.RequestFactory$Builder.build(RequestFactory.java:213)
	at retrofit2.RequestFactory.parseAnnotations(RequestFactory.java:67)
	at retrofit2.ServiceMethod.parseAnnotations(ServiceMethod.java:26)
	at retrofit2.Retrofit.loadServiceMethod(Retrofit.java:202)
	at retrofit2.Retrofit$1.invoke(Retrofit.java:160)
	at jdk.proxy2/jdk.proxy2.$Proxy27.listRunSteps(Unknown Source)
	at com.theokanning.openai.service.OpenAiService.listRunSteps(OpenAiService.java:494)

```

and the test code (GptService.getService() return the OpenAiService already initialized)

```java
        @Test
	void test_execute_Assistant() throws InterruptedException
	{
		var service=gptService.getService();
		var assistant = service.retrieveAssistant("<Assistant ID>");
		
		var message=MessageRequest.builder()
				.role(ChatMessageRole.USER.value())
				.content("QUESTION: explain the Strategy pattern")
				.build();
		
		var threadRequest=ThreadRequest.builder()
				.messages(List.of(message))
				.build();
		
		
		var thread_run = CreateThreadAndRunRequest.builder()
				.assistantId(assistant.getId())
				.model(assistant.getModel())
				.thread(threadRequest)
				.build();
		
		var run = service.createThreadAndRun(thread_run);
		// run status
		// queued|in_progress|completed| failed ?
		var running=Set.of("queued","in_progress");
		while (running.contains(run.getStatus()))
		{
			log.info(run.toString());
			Thread.currentThread().sleep(Duration.ofMillis(1000));
			run =  service.retrieveRun(run.getThreadId(), run.getId());
		}
		log.info("run {}",run.toString());
		var runStep=service.listRunSteps(run.getThreadId(),run.getId(),ListSearchParameters.builder().build());
		log.info("thread {}",runStep.toString());
		
	}

```

**Comment by Aelentel** [comment]
just adding that i got the response nonetheless by using listMessage api, so if it can help other ppl, here's the code i'm using.

```java
		var messageList=service.listMessages(run.getThreadId());
		String response=
				messageList.getData()
					.stream()
					.filter(t -> ChatMessageRole.ASSISTANT.value().equals(t.getRole()))
					.flatMap(t -> t.getContent().stream())
					.filter(t -> "text".equals(t.getType()))
					.map(t -> t.getText().getValue())
					.findFirst()
					.orElse(null);
```

**Comment by vacuityv** [comment]
@Aelentel  I will check this and if I know why I wiill try to fix it

**Comment by vacuityv** [comment]
@Aelentel  I will fix this.

**Comment by vacuityv** [comment]
A pr here: https://github.com/TheoKanning/openai-java/pull/414

**Comment by TheoKanning** [comment]
Fixed in 0.18.2
